### PR TITLE
[CBR-494] Per node keep-alive timers

### PR DESCRIPTION
**Pull request by karknu**
The keep-alive timer where incorrectly shared among all
nodes a client subscribed to which rendered them ineffective
during rolling restarts.

## Linked issue
https://iohk.myjetbrains.com/youtrack/issue/CBR-494

## Type of change

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## QA Steps

0. Setup two private, p-0 and p-1, relays and one public relay. The public relay should subscribe to both private relay at the same time (dynamic-subscribe: [ [ { "host":"p-0.cardano"} ], [ {"host":"p-1.cardano"} ] ] ).
1. Start a fresh block sync from a client that connects to the public relay.
2. Restart p-0
3. Wait 4 minutes
4. Restart p-1
5. Verify that it was possible to restart both nodes without any visible interruption in the client.

## How to merge

Send the message `bors r+` to merge this PR. For more information, see
[`docs/how-to/bors.md`](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/how-to/bors.md).

### CODE REVIEW

File: lib/src/Pos/Diffusion/Full/Block.hs [hunk 1]
```diff
@@ -593,7 +592,7 @@ blockListeners
     -> ProtocolConstants
     -> Word
     -> OQ.OutboundQ pack NodeId Bucket
-    -> Timer -- ^ Keepalive timer
+    -> (NodeId -> IO ()) -- ^ Keepalive timer
```
Comment (coot) [inline lib/src/Pos/Diffusion/Full/Block.hs]: Why not add a newtype wrapper `ResetNodeTimer` for this.  It will convey the meaning of this `IO` action.

Comment (tatyanavych) [reply lib/src/Pos/Diffusion/Full/Block.hs]: We completed cardano-sl release/2.0.0. I  will check whether we need to have this as a hotfix.

**Comment by karknu** [comment]
This PR will not go into 2.0.0.

**Pull request merged**